core/src/main/java/org/apache/brooklyn/core/mgmt/persist/BrooklynMementoPersisterToObjectStore.java

```diff
@@ -294,7 +294,7 @@ public void visit(BrooklynObjectType type, String id, String contentsSubpath) th
                     exceptionHandler.onLoadMementoFailed(type, "memento "+id+" read error", e);
                 }
                 
-                String xmlId = (String) XmlUtil.xpath(contents, "/"+type.toCamelCase()+"/id");
+                String xmlId = (String) XmlUtil.xpathHandlingIllegalChars(contents, "/"+type.toCamelCase()+"/id");
```

Comment:
Would `xpathHandlingIllegalChars` escape mechanism will be appropariate for filenames?
Ideally if a function like `File.escapeSpecialChars` exists then this should look like `File.escapeSpecialChars(XmlUtil.xpathHandlingIllegalChars(contents, "/"+type.toCamelCase()+"/id"))`